### DISCUSSION TITLE
Is it possible to access env without using c.env?

### DISCUSSION THREAD

**Discussion by juniorforlife** [honojs/hono#2764] [Q&A] [answered]
I find it strange that I can only access the env variables inside a route or middleware handler via a context. Why can't I just use process.env or env helper without passing in the context?

**Comment by NicoPlyley** [comment, marked as answer]
This is only the case when using Cloudflare Workers / Pages. This is not a limitation on Hono, but on how Cloudflare provides the environment variables. Here is how it is being handled in the background:

```ts
export default {
  async fetch(request, env) {
    myHonoApp.fetch(request, env)
};
```
On each request made to a worker, the environment variables are also sent, that is why they are only accessible in context, they are not made available globally.

**Reply by juniorforlife** [reply]
@NicoPlyley how about AWS Lambda? Do I need to use c.env in Lambda or I can access it the same way as Node?

**Reply by NicoPlyley** [reply]
It appears so, I have not ever used Lambda

https://docs.aws.amazon.com/lambda/latest/dg/configuration-envvars.html#configuration-envvars-retrieve

**Reply by juniorforlife** [reply]
thanks for your help!

**Comment by sp42** [comment]
I got this problem too. How can I get these env without c.env, that means, why not just using global var or static API?

**Reply by NicoPlyley** [reply]
Check my answer above on why. This is how CF Workers is setup, not a limitation on Hono. There is nothing Hono can do to make them globally available like in node.
